components/kserve/serverless_setup.go

```diff
@@ -15,55 +17,55 @@ const (
 	templatesDir            = "templates/serverless"
 )
 
-func (k *Kserve) configureServerlessFeatures(s *feature.FeaturesInitializer) error {
-	var rootDir = filepath.Join(feature.BaseOutputDir, s.DSCInitializationSpec.ApplicationsNamespace)
-	if err := feature.CopyEmbeddedFiles(templatesDir, rootDir); err != nil {
-		return err
-	}
+func (k *Kserve) configureServerlessFeatures(dscispec *dsci.DSCInitializationSpec, origin featurev1.Origin) feature.DefinedFeatures {
+	return func(s *feature.FeaturesInitializer) error {
+		var rootDir = filepath.Join(feature.BaseOutputDir, s.DSCInitializationSpec.ApplicationsNamespace)
+		if err := feature.CopyEmbeddedFiles(templatesDir, rootDir); err != nil {
+			return err
+		}
 
-	servingDeployment, err := feature.CreateFeature("serverless-serving-deployment").
-		For(s.DSCInitializationSpec).
-		Manifests(
-			path.Join(rootDir, templatesDir, "serving-install"),
-		).
-		WithData(PopulateComponentSettings(k)).
-		PreConditions(
-			serverless.EnsureServerlessOperatorInstalled,
-			serverless.EnsureServerlessAbsent,
-			servicemesh.EnsureServiceMeshInstalled,
-			feature.CreateNamespaceIfNotExists(knativeServingNamespace),
-		).
-		PostConditions(
-			feature.WaitForPodsToBeReady(knativeServingNamespace),
-		).
-		Load()
-	if err != nil {
-		return err
-	}
-	s.Features = append(s.Features, servingDeployment)
+		servingDeployment, err := feature.CreateFeature("serverless-serving-deployment").
+			For(dscispec, origin).
```

Comment:
i must be confused, but should not serverless has DSC from origin? 
(not related to this PR but the old code base) 

e.g Type: featurev1.ComponentType, here should set to use 
`DSCType OwnerType = "DSC"`
not to use `DSCIType OwnerType = "DSCI"`

Reply:
ok, i think kinda understand it here. 
it is either DSCI + DSCI CR's name
or  Component + component's name.

Reply:
That's right, but it's kinda blurry. For now, maybe we can rework it to not have one `For` for both, as it doesn't make sense. Lemme work with @cam-garrison so we can improve the API and close this PR.